src/topology/algebra/ring.lean

```diff
@@ -106,7 +102,7 @@ variables [topological_ring α]
   makes the product into a topological ring. -/
 instance {β : Type*}
   [ring β] [topological_space β] [topological_ring β] : topological_ring (α × β) :=
-{ continuous_neg := continuous_neg }
+{ }
```

Comment:
Is there a reason you're able to remove this line but not line 109?

Reply:
Sometimes Lean fails to unify different instances on pi types. I'm not sure what exactly is happening here but it fails to accept, e.g.,  `to_topological_add_group := pi.topological_add_group`.

Reply:
I have no idea why it fails to unify `topological_add_group` instances but accepts `continuous_neg`.